src/wallet/wallet.cpp

```diff
@@ -382,11 +377,6 @@ std::shared_ptr<CWallet> CreateWallet(WalletContext& context, const std::string&
     const SecureString& passphrase = options.create_passphrase;
 
     if (wallet_creation_flags & WALLET_FLAG_DESCRIPTORS) options.require_format = DatabaseFormat::SQLITE;
-    else {
```

Comment:
In e66e9ee9f19413c93d8523afdb8f9f3e07b818ec "wallet, refactor: Remove Legacy warnings and errors"

I'd prefer to at least preserve enforcement that the wallet being created is a descriptor wallet. Perhaps we could add an `Assert()` or `Assume()` for the flag.

Reply:
I did that earlier but got a failure and I thought it was related with migration and rolled it back without checking it properly. I'll add it back.